File: src/main/java/com/tabnine/logging/TabnineFilter.java
package com.tabnine.logging;

import org.apache.log4j.spi.Filter;
import org.apache.log4j.spi.LoggingEvent;

public class TabnineFilter extends org.apache.log4j.spi.Filter {

    @Override
    public int decide(LoggingEvent loggingEvent) {
        return loggingEvent.getLoggerName().startsWith("#com.tabnine") ? Filter.ACCEPT : DENY;
    }
}

File: src/main/java/com/tabnine/logging/TabnineFilter.kt
package com.tabnine.logging

import org.apache.log4j.spi.Filter
import org.apache.log4j.spi.LoggingEvent

class TabnineFilter : Filter() {
    override fun decide(loggingEvent: LoggingEvent): Int {
        return if (loggingEvent.loggerName.startsWith("#com.tabnine")) ACCEPT else DENY
    }
}
